### PR TITLE
[Intel MKL] Enabling MKL Conv2D BWD in eager mode

### PR DESCRIPTION
**Pull request by mahmoud-abuzaina**
This PR depends on #30401. So please review and merge #30401 first.

### CODE REVIEW

**Review by penpornk — CHANGES_REQUESTED** [review]
Thank you for your PR! 
I have an overall question: Why do we need an additional op for the eager version? Can't we just add `eager` as a parameter to `_MklConv2DBackpropFilter` and `_MklConv2DBackpropInput`?

**Comment by gbaned** [comment]
@mahmoud-abuzaina Did you get a chance to look on reviewer comments? Please let us know on the update. Thanks!

File: tensorflow/core/kernels/mkl_conv_grad_input_ops.cc [hunk 1]
```diff
@@ -429,9 +432,12 @@ class MklConvCustomBackpropInputOp
                                      bwd_diff_src_dims, bwd_diff_src_format);
       TensorShape diff_src_tf_shape;
       diff_src_tf_shape.AddDim(diff_src_pd.get_size() / sizeof(T));
+      if (eager_mode) {
```
[the remark this answers is not in the document]
Comment (gzmkl) [reply tensorflow/core/kernels/mkl_conv_grad_input_ops.cc]: Sure, I will do

**Comment by gzmkl** [comment]
@gbaned: I am sorry for the late response. After handling the related PR "eagle mode Conv fwd",
I can deal with the "bwd" one. I may not answer all questions/comments since I am covering
the original developer. But I will try my best.

File: tensorflow/core/kernels/mkl_conv_grad_input_ops.cc (hunk 1 above)
[the remark this answers is not in the document]
Comment (gzmkl) [reply tensorflow/core/kernels/mkl_conv_grad_input_ops.cc]: Done with code change

**Review by penpornk — APPROVED** [review]
Thank you for the fix!

**Review by penpornk — CHANGES_REQUESTED** [review]
Oops. Just realized my question about adding `eager` as an op attribute instead of template parameter hasn't been answered yet. Taking the approval back for now. (Probably have to wait for the FWD PR https://github.com/tensorflow/tensorflow/pull/30401)

**Review by penpornk — APPROVED** [review]

**Review by alextp — APPROVED** [review]
@penpornk I think in the future we shouldn't need special eager-owners approval for these changes; now that the mechanism is agreed upon these should be safe.

**Comment by penpornk** [comment]
@alextp Got it. Then I can handle MKL-related, incremental eager PRs alone and will request reviews from eager owners when there are major changes.

**Comment by gzmkl** [comment]
@penpornk  Hi Penporn, please help to get this one merged into 2.0. Thanks!   -GZ

**Comment by penpornk** [comment]
@gzmkl I'm working on getting it merged in the master branch. :) After that, I'll see if we can cherrypick this into 2.0. Same for https://github.com/tensorflow/tensorflow/pull/31311.

**Pull request merged**